modules/nf-core/nonpareil/tests/main.nf.test

```diff
@@ -0,0 +1,87 @@
+nextflow_process {
+
+    name "Test Process NONPAREIL"
+    script "../main.nf"
+    process "NONPAREIL"
+    config "./nextflow.config"
+    tag "modules"
+    tag "modules_nfcore"
+    tag "nonpareil"
+
+    test("candidatus_portiera_aleyrodidarum - gzipped") {
+
+        when {
+            params {
+                // define parameters here. Example:
+            // outdir = "tests/results"
+            }
+            process {
+                """
+                input[0] = [
+                    [ id:'test', single_end:false ], // meta map
+                    file(params.test_data['candidatus_portiera_aleyrodidarum']['illumina']['test_1_fastq_gz'], checkIfExists: true)
+                ]
+                input[1] = 'fastq'
+                input[2] = 'kmer'
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert process.out.npa.get(0).get(1) ==~ ".*/test.npa" },
+                { assert process.out.npc.get(0).get(1) ==~ ".*/test.npc" },
+                { assert path(process.out.npo.get(0).get(1)).getText().contains("# @impl: Nonpareil") },
+                { assert path(process.out.npl.get(0).get(1)).md5 == "be95453a272a9f685d5695d2b391fde8" },
```

Comment:
Maybe also add these to the snapshot to reduce the maintenance on them in the future :)

Reply:
Just thinking in the comments here. I think we need to make this a requirement. If there's a reason not to, then a comment with a note about why it's not in the snapshot?

Reply:
I agree!!